pkg/scaffold/resource/resource_test.go

```diff
@@ -94,38 +94,49 @@ var _ = Describe("Resource", func() {
 		It("should default the Resource by pluralizing the Kind", func() {
 			instance := &Resource{Group: "crew", Kind: "FirstMate", Version: "v1"}
 			Expect(instance.Validate()).To(Succeed())
+			Expect(instance.Init()).To(Succeed())
+			Expect(instance.Plural).To(Equal("firstmates"))
 			Expect(instance.Resource).To(Equal("firstmates"))
 
 			instance = &Resource{Group: "crew", Kind: "Fish", Version: "v1"}
 			Expect(instance.Validate()).To(Succeed())
+			Expect(instance.Init()).To(Succeed())
+			Expect(instance.Plural).To(Equal("fish"))
 			Expect(instance.Resource).To(Equal("fish"))
 
 			instance = &Resource{Group: "crew", Kind: "Helmswoman", Version: "v1"}
 			Expect(instance.Validate()).To(Succeed())
+			Expect(instance.Init()).To(Succeed())
+			Expect(instance.Plural).To(Equal("helmswomen"))
 			Expect(instance.Resource).To(Equal("helmswomen"))
 		})
 
 		It("should allow Cat as a Kind", func() {
 			instance := &Resource{Group: "crew", Kind: "Cat", Version: "v1"}
 			Expect(instance.Validate()).To(Succeed())
+			Expect(instance.Init()).To(Succeed())
+			Expect(instance.Plural).To(Equal("cats"))
```

Comment:
Has no need to add the plural since the {{ .Resource.Resource  }} has been using to keep it.

Reply:
`kubebuilder create webhook` (and its v1 equivalent `kubebuilder alpha webhook`) offer a `--resource` flag that replaces the default behavior of storing the plural in the `Resource.Resource` field. Different scaffolding files were using a `.Plural` field to ensure that it doesn't use a user-specified resource. Basically I moved the file `.Plural` field into the `Resource.Plural` to avoid calling `flect.Pascalize` and `strings.ToLower` in multiple places.

Reply:
I  understand, however, IMO the only refactory over it that we can do is use the ` {{ .Resource.Resource }} ` and has no need to add the check in the tests.

Reply:
Tests have to be unaware of implementation. Both are expected to be the same in case we do not provide the `--resource` flag, so we should test both. Imagine someone in a later commit replaces plural with a PascalCase version without noticing. We would have a bug that tests are not catching. More tests = better tests always.

Reply:
The pascalized should be removed and we should use the k8s validation to check the Kind values. See : https://github.com/kubernetes-sigs/kubebuilder/pull/1133

Reply:
#1133 still enforces a letter to be in first position, so my argument still aplies.

Reply:
The PascalCase was just an example. Including both fields in tests let us be sure that the behavior won't change without noticing. If we don't include Plural in the tests this may change in a later comit resulting in bugs in the scaffolded files that tests will not be catching.